src/category_theory/closed/monoidal.lean

```diff
@@ -48,4 +65,171 @@ def unit_closed {C : Type u} [category.{v} C] [monoidal_category.{v} C] : closed
       hom_equiv_naturality_left_symm' := λ X' X Y f g,
       by { dsimp, rw left_unitor_naturality_assoc } } } }
 
+variables (A B : C) {X X' Y Y' Z : C}
+
+variables [closed A]
+
+/-- This is the internal hom `A ⟹ -`. -/
+def ihom : C ⥤ C :=
+(@closed.is_adj _ _ _ A _).right
```

Comment:
This will give really bad defeqs in interesting cases like `Module R`, right?
Shouldn't have have a `has_ihom` typeclass? And a "predicate" (read: adjunction) relating it to the monoidal structure in `monoidal_closed`?

Or should we rather tackle that refactor in a follow-up PR?

Reply:
Good point! I hadn't run into this problem yet in using this, but I agree it would be good to do. I'll add a comment here, proposing that this be done later.